atomix/cluster/src/test/java/io/atomix/raft/partition/StepDownIfNotPrimaryTest.java

```diff
@@ -0,0 +1,103 @@
+    // given
+    final var primaryMemberId = new MemberId("2");
+    final var partitionId = new PartitionId("group", 1);
+    final var raftPartitionConfig = new RaftPartitionConfig();
+    final var partition =
+        new RaftPartition(
+            partitionId,
+            new RaftPartitionGroupConfig().setPartitionConfig(raftPartitionConfig),
+            tempDir.toFile());
+    final var raftPartitionServer = Mockito.mock(RaftPartitionServer.class);
+    Mockito.when(raftPartitionServer.getMemberId()).thenReturn(new MemberId("1"));
+
+    // To avoid having to start a server, just mock one.
+    FieldUtils.writeField(partition, "server", raftPartitionServer, true);
+
+    // when -- enabling priority election and marking a different member as primary
+    raftPartitionConfig.setPriorityElectionEnabled(true);
+    partition.setMetadata(
+        new PartitionMetadata(partitionId, Set.of(), Map.of(), 1, primaryMemberId));
+
+    // then -- current member should step down
+    Assertions.assertThat(partition.shouldStepDown()).isTrue();
+  }
+
+  @Test
+  void shouldNotStepDownIfPrimary(@TempDir Path tempDir) throws IllegalAccessException {
+    // given
+    final var partitionId = new PartitionId("group", 1);
+    final var raftPartitionConfig = new RaftPartitionConfig();
+    final var partition =
+        new RaftPartition(
+            partitionId,
+            new RaftPartitionGroupConfig().setPartitionConfig(raftPartitionConfig),
+            tempDir.toFile());
+
+    final var raftPartitionServer = Mockito.mock(RaftPartitionServer.class);
+    Mockito.when(raftPartitionServer.getMemberId()).thenReturn(new MemberId("1"));
+
+    // To avoid having to start a server, just mock one.
+    FieldUtils.writeField(partition, "server", raftPartitionServer, true);
+
+    // when -- enabling priority election and marking the current member as primary
+    final var primaryMemberId = new MemberId("1");
+    raftPartitionConfig.setPriorityElectionEnabled(true);
+    partition.setMetadata(
+        new PartitionMetadata(partitionId, Set.of(), Map.of(), 1, primaryMemberId));
+
+    // then -- current member should not step down
+    Assertions.assertThat(partition.shouldStepDown()).isFalse();
+  }
+
+  @Test
+  void shouldNotStepDownIfPriorityElectionDisabled(@TempDir Path tempDir)
+      throws IllegalAccessException {
+    // given
+    final var partitionId = new PartitionId("group", 1);
+    final var raftPartitionConfig = new RaftPartitionConfig();
+    final var partition =
+        new RaftPartition(
+            partitionId,
+            new RaftPartitionGroupConfig().setPartitionConfig(raftPartitionConfig),
+            tempDir.toFile());
+    final var raftPartitionServer = Mockito.mock(RaftPartitionServer.class);
+    Mockito.when(raftPartitionServer.getMemberId()).thenReturn(new MemberId("1"));
+
+    // To avoid having to start a server, just mock one.
+    FieldUtils.writeField(partition, "server", raftPartitionServer, true);
```

Comment:
I'm not sure if it'd make the test easier tbh. We'd still want `memberId` to be a private field. 
So instead of 
```java
    final var mockRaftPartitionServer = Mockito.mock(RaftPartitionServer.class);
    Mockito.when(mockRaftPartitionServer.getMemberId()).thenReturn(new MemberId("1"));
    FieldUtils.writeField(partition, "server", mockRaftPartitionServer, true);
```
we'd do 
```java
    FieldUtils.writeField(partition, "memberId", new MemberId("1"), true);
```

Additionally, we'd need to keep `memberId` and `server` in `RaftPartitionServer` in sync.